### ISSUE TITLE
Evaluate None token passed to `context.detach`

### ISSUE THREAD

**Issue by xrmx** [open-telemetry/opentelemetry-python-contrib#3351]
Evaluate the users of the context api to check that the callers of the `detach` function are passing a token that is not None.
This can be checked by inspection or by throwing pyright at the instrumentation code and stasrt fixing any typing issue.

For finding entries search for `detach(token)`

**Comment by aryabharat** [comment]
@xrmx can i pick this ?

**Comment by xrmx** [comment]
@aryabharat When you are ready to work on this just state so in a comment and we'll consider this assigned to you for one week.

**Comment by aryabharat** [comment]
@xrmx sure.
should we handle None check where ever we `context.detach()` is called?

**Comment by xrmx** [comment]
> [@xrmx](https://github.com/xrmx) sure. Should we handle the None check wherever context.detach() is called?"

yes
